enterprise/reporting/reporting_test.go

```diff
@@ -159,81 +159,183 @@ var _ = Describe("Reporting", func() {
 				},
 			},
 		}
+		conf := config.New()
+		configSubscriber := newConfigSubscriber(logger.NOP)
+		reportHandle := NewDefaultReporter(context.Background(), conf, logger.NOP, configSubscriber, stats.NOP)
 
-		expectedResponse := []*types.Metric{
-			{
-				InstanceDetails: types.InstanceDetails{
-					WorkspaceID: "some-workspace-id",
-				},
-				ConnectionDetails: types.ConnectionDetails{
-					SourceID:         "some-source-id",
-					DestinationID:    "some-destination-id",
-					TransformationID: "some-transformation-id",
-					TrackingPlanID:   "some-tracking-plan-id",
-				},
-				PUDetails: types.PUDetails{
-					InPU: "some-in-pu",
-					PU:   "some-pu",
-				},
-				ReportMetadata: types.ReportMetadata{
-					ReportedAt: 28017690 * 60 * 1000,
-				},
-				StatusDetails: []*types.StatusDetail{
-					{
-						Status:         "some-status",
-						Count:          3,
-						ViolationCount: 5,
-						StatusCode:     200,
-						SampleResponse: "",
-						SampleEvent:    []byte(`{}`),
-						ErrorType:      "",
+		It("Should provide aggregated reports when batch size is 1", func() {
```

Comment:
we should add 1 more test for having multiple entries for same identifier in the expected response. same when batch size is more than 1.

Reply:
added this test of batch size > 1. for batch size = 1, this is covered by `Should provide aggregated reports when batch size is 1`